Managing hosts adding network interfaces (#3431)
theforeman/foreman-documentation: guides/common/modules/ref_network-interface-configuration-options.adoc

[id="network_interface_configuration_options"]
= Network interface configuration options

When adding a network interface, you need to specify several configuration options. The following lists provide information on the relevant options for the different types of interfaces.

.Physical interface settings

*Device Identifier*:: The identifier is used to specify this physical interface when creating bonded interfaces, VLANs, and aliases.

*DNS name*:: {Project} saves this name in the {SmartProxyServer} associated with the selected domain (the "DNS A" field) and {SmartProxyServer} associated with the selected subnet (the "DNS PTR" field).
A single host can therefore have several DNS entries.

*Domain*:: Specifies the domain associated with the network interface. To create and manage domains, navigate to *Infrastructure* > *Domains*.

*Subnet*:: Specifies the subnet associated with the network interface. To create and manage subnets, navigate to *Infrastructure* > *Subnets*.

*Managed* interface:: If the interface is managed, configuration is pulled from the associated {SmartProxyServer} during provisioning, and DNS and DHCP entries are created.

> If using Kickstart provisioning, a configuration file is automatically created for the interface.

Reviewer: ```suggestion
If using installer-based provisioning, a configuration file is automatically created for the interface.
```
Kickstart is installer configuration specific to EL, but I think it applies to any OS installer. I'd go with a more generic term to make it more upstream friendly.

Reviewer: @maximiliankolb Can you or someone from your team confirm it applies to other OSs as well?

Reviewer: In the context of Foreman, installer-based provisioning should be considered automated. Anaconda+Kickstart is for EL, but Debian/Ubuntu uses a different installer+config. But Foreman generates configuration for Debian/Ubuntu as well, it just isn't called Kickstart.
Then perhaps a better term would be:
```suggestion
If you use provisioning with installer automation, a configuration file is automatically created for the interface.
```

Reviewer: @stejskalleos Would that ^^ work?

Reviewer 2: If, in this context, we expect the installer workflow always to be automated, then LGTM.

Reviewer 3: Using a OS-neutral term would be appreciated, yes.

*Primary* interface:: The DNS name from the primary interface is used as the host portion of the FQDN.

> *Provision*:: TFTP boot takes place using the provisioning interface.

Reviewer: ```suggestion
*Provision*:: Network boot uses the provisioning interface.
```

Reviewer: @stejskalleos Is is this change acceptable?

Reviewer 2: LGTM
